megablocks/layers/moe.py

```diff
@@ -98,7 +100,8 @@ def batched_load_balancing_loss(args : Arguments):
         args.moe_top_k
     )
     scale = scale_numerator / scale_denominator
-    return scale * torch.dot(tokens_per_expert, expert_scores)
+    zloss = (torch.log(torch.exp(expert_logits).sum(dim=-1)) ** 2).sum() / scale_denominator
```

Comment:
Implemented like this, the log-sum-exp operation is numerically unstable, this should use `torch.logsumexp` instead